plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/refactoring/introduce/introduceConstant/IntroduceConstUtil.kt

```diff
@@ -0,0 +1,81 @@
+// Copyright 2000-2021 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.jetbrains.kotlin.idea.refactoring.introduce.introduceConstant
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.caches.resolve.resolveToCall
+import org.jetbrains.kotlin.idea.structuralsearch.visitor.KotlinRecursiveElementVisitor
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+
+/**
+ * [FqName] of operators which can be used in a kotlin const expression.
+ */
+private val constOperatorNames = listOf("Byte", "Short", "Int", "Long", "UByte", "UShort", "UInt", "ULong")
+    .flatMap {
```

Comment:
As well there are many others like binary operators `shl` (e.g. `1 shl 10` that is 1024), `and`, `or` etc - that is quite often used as constants

Reply:
Automatically included by using `ConstantExpressionEvaluator`.
I added a test for this